lua/wikis/valorant/MatchGroup/Input/Custom/MatchPage.lua

```diff
@@ -0,0 +1,1986 @@
+---
+-- @Liquipedia
+-- wiki=valorant
+-- page=Module:MatchGroup/Input/Custom/MatchPage
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+
+local CustomMatchGroupInputMatchPage = {}
+
+---@class valorantMatchDataExtended: valorantMatchData
+---@field matchid string
+---@field vod string?
+---@field finished boolean
+
+---@param mapInput {matchid: string?, reversed: string?, vod: string?, region: string?}
+---@return dota2MatchDataExtended|table
+function CustomMatchGroupInputMatchPage.getMap(mapInput)
+	-- If no matchid is provided, assume this as a normal map
+	if not mapInput or not mapInput.matchid then
+		return mapInput
+	end
+	assert(mapInput.matchid, 'Numeric matchid expected, got ' .. mapInput.matchid)
+	assert(mapInput.region, 'Region is required ')
+
+	--local map = mw.ext.valorantdb.getDetails(mapInput.matchid, mapInput.region, Logic.readBool(mapInput.reversed))
+	local map = CustomMatchGroupInputMatchPage.getMockData()
+
+	assert(map and type(map) == 'table' and map.matchInfo, mapInput.matchid .. ' could not be retrieved.')
+	-- Let's shift the array to start from 1
+	if map.players and type(map.players) == 'table' and map.players[0] then
+		local newTeams = {}
+		for i, team in pairs(map.players) do
+			local newPlayers = {}
+			for j, player in pairs(team) do
+				newPlayers[j + 1] = player
+			end
+
+			newTeams[i + 1] = team
+			newTeams[i + 1].players = newPlayers
+		end
+		map.players = newTeams
+	end
+	if map.roundDetails and type(map.roundDetails) == 'table' and map.roundDetails[0] then
+		local newRoundDetails = {}
+		for i, roundDetail in pairs(map.roundDetails) do
+			newRoundDetails[i + 1] = roundDetail
+		end
+		map.roundDetails = newRoundDetails
+	end
+	if map.teams and type(map.teams) == 'table' and map.teams[0] then
+		local newTeams = {}
+		for i, team in pairs(map.teams) do
+			newTeams[i + 1] = team
+		end
+		map.teams = newTeams
+	end
+
+	---@cast map valorantMatchDataExtended
+	map.matchid = mapInput.matchid
+	map.vod = mapInput.vod
+	map.finished = true
+
+	return map
+end
+
+---@param map table
+---@param opponentIndex integer
+---@return table[]?
+function CustomMatchGroupInputMatchPage.getParticipants(map, opponentIndex)
+	local teamPlayers = map['players'][opponentIndex]
+	if not teamPlayers then return end
+
+	local function fetchLpPlayer(playerId)
+		--- TODO
```

Comment:
assume this is oos for this PR and will be done in a follow up PR

Reply:
Yeah, can't do that yet, not 100% what identifiers I get from the API when it's made, need the real API contract first instead of this dummy data dump